cognite/client/data_classes/assets.py

```diff
@@ -259,7 +259,10 @@ def _retrieve_related_resources(self, resource_list_class, resource_api):
             resource_api.list, tasks, resource_api._config.max_workers
         ).results
         resources = resource_list_class([])
+        file_ids = []
         for res in res_list:
+            res = resource_list_class([item for item in res if item.id not in file_ids])
```

Comment:
I would prefer a new variable for this, `new_res` or something, rather than overwriting the previous value, but maybe I'm being overly against the imperative programming style. ;)